### PR TITLE
Multimodal factorized bilinear pooling for VQA

### PR DESCRIPTION
**Pull request by Drajan**
Image and Text feature fusion as described in - https://arxiv.org/abs/1708.03619

### CODE REVIEW

File: ptp/components/models/vqa/multimodal_factorized_bilinear_pooling.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        latent_img = self.dropout(self.image_encodings_ff(enc_img)) # [48, 512]
+        latent_q =  self.dropout(self.question_encodings_ff(enc_q)) # [48, 512]
+
+        # Element-wise mutliplication of image and question encodings
+        enc_z = latent_img * latent_q # [48, 512]
+        # Dropout regularization
+        enc_z = self.dropout(enc_z)
+        enc_z = enc_z.view(enc_z.size(0), self.latent_size, self.factor) # [48, 256, 2]
+        # Sum pooling
+        enc_z = enc_z.sum(2) # [48, 256]
```
Comment (tkornut) [inline ptp/components/models/vqa/multimodal_factorized_bilinear_pooling.py]: in the paper they eq. (4) introduces k=2, if I am right this "k" is self.factor ("pooling_factor").

Should that be used in here?

Comment (Drajan) [reply ptp/components/models/vqa/multimodal_factorized_bilinear_pooling.py]: would it be better to change line 120 to
enc_z = torch.sum(enc_z, dim=self.factor) ?

Comment (tkornut) [reply ptp/components/models/vqa/multimodal_factorized_bilinear_pooling.py]: you are right! :)
